File: medplum-java/medplum-model/src/main/java/com/fhir/Device_Property.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableDevice_Property.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface Device_Property {
  Optional<List<CodeableConcept>> valueCode();

  Optional<List<Quantity>> valueQuantity();

  Optional<List<Extension>> modifierExtension();

  Optional<List<Extension>> extension();

  CodeableConcept type();

  Optional<String> id();

  static ImmutableDevice_Property.TypeBuildStage builder() {
    return ImmutableDevice_Property.builder();
  }
}

File: medplum-typescript/src/model/fhir/Device_Property.ts
import { CodeableConcept } from "./CodeableConcept";
import { Extension } from "./Extension";
import { Quantity } from "./Quantity";

export interface Device_Property {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    type: CodeableConcept;
    valueQuantity: Quantity[] | null | undefined;
    valueCode: CodeableConcept[] | null | undefined;
}
